### PR TITLE
Fix rectangular patches to be properly transformed on polar axes

### PR DESCRIPTION
**Pull request by Tuan333**
## PR Summary
Rectangle patch not transformed correctly in polar plot #8521. The bug fix tries to specify the number of _interpolation_steps for Path._unit_rectangle in the _AxesBase.add_patch() function. It appears that 'polar' transforms require more than the default _interpolation_steps == 1.

More test cases can be generated upon request

https://github.com/matplotlib/matplotlib/issues/8521
## PR Checklist

- [x] Has Pytest style unit tests
- [x] Code is PEP 8 compliant 
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/whats_new.rst if major new feature
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by phobson** [comment]
This PR looks good. After a little more review happens, would you mind squashing these commits down to a single commit? If not we can take care of it when it's merged.

File: lib/matplotlib/tests/test_proj_on_rect_patch.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import math
+import matplotlib.pyplot as plt
+from matplotlib import patches
+import matplotlib
+from matplotlib.testing.decorators import image_comparison
+
+
+@image_comparison(baseline_images=['polar_proj'], extensions=['png'])
+def test_adding_rectangle_patch():
+    fig = plt.figure(figsize=(640/80, 640/80), dpi=80)
```
Comment (dstansby) [inline lib/matplotlib/tests/test_proj_on_rect_patch.py]: I don't think you need to specify the figsize or dpi, just `plt.figure()` should be fine.

File: lib/matplotlib/tests/test_proj_on_rect_patch.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    ax = fig.add_subplot(111, projection='polar')
+
+    # add quadrant as example
+    ax.add_patch(
+        patches.Rectangle(
+            (0, 1), width=math.pi * 0.5, height=0.5
```
Comment (dstansby) [inline lib/matplotlib/tests/test_proj_on_rect_patch.py]: To be consistent with the rest of the tests, this should use numpy and be `np.pi`

Comment (Tuan333) [inline lib/matplotlib/tests/test_proj_on_rect_patch.py]: Thanks for the input. Will fix this as soon as I get home today from work.

**Comment by QuLogic** [comment]
I'm not sure we need a separate file for this either; it should be able to fit in `test_axes.py`, I think.

**Comment by QuLogic** [comment]
Oh, I see @dstansby suggested `test_patches.py`, which seems good too.

**Comment by Tuan333** [comment]
@phobson Could you please kindly give me some pointers on how to squash the commits together? I don't normally use git, and have not found anything useful on the internet :)

**Comment by phobson** [comment]
@dstansby can you take another look at this?

**Pull request merged**